src/lib.rs

```diff
@@ -42,456 +42,417 @@
-                let [r, g, b] = COLORS_24BIT.get(code)?;
+        let color_map =
+            colors(self.depth).expect("at this point depth should only be TTY, Low or Medium");
 
-                return Some(fill_rgb_template(template, r, g, b));
-            }
+        let mapped = color_map(color).expect("the regex should match only valid color codes");
 
-            let color_map = COLORS[&(self.depth as u8)];
-            let mut value = color_map.get(code)?.to_string();
+        // low bit depths use different way of specifying background
+        let value = if bg && self.depth <= Depth::Low {
+            (mapped
+                .parse::<u8>()
+                .expect("color tables should contain valid numbers")
+                + 10)
+                .to_string()
+        } else {
+            mapped.to_string()
+        };
 
-            if bg && self.depth <= Depth::Low {
-                value = (value.parse::<u8>().ok()? + 10).to_string()
-            };
+        fill_template(templater(self.depth), &value)
+    }
 
-            Some(fill_template(template, &value))
-        }
+    /// Writes the prompt to stdout, then reads a line from input,
+    /// and returns it (excluding the trailing newline).
+    pub fn input(&self, prompt: &str) -> std::io::Result<String> {
+        print!("{}", self.convert(prompt));
+        stdout().flush()?;
+
+        let mut inp = String::new();
+        stdin().read_line(&mut inp)?;
+        Ok(inp.trim_end().to_owned())
     }
 
     /// Resets the formatting back to the default.
-    pub fn reset(&self) {
-        print!("{}", self.convert(format!("{}r", self.marker)));
+    pub fn reset() {
+        print!("\x1b[0m");
+    }
+
+    /// Clears the current line
+    pub fn clear_line() {
+        print!("\x1b[2K");
+    }
+
+    /// Clears the screen
+    pub fn clear_screen() {
+        print!("\x1b[2J");
+    }
+
+    /// Escape all format markers in a string
+    ///
+    /// # Example
+    /// ```rust
+    /// # use dahlia::Dahlia;
+    /// let d = Dahlia::default();
+    /// let str = d.escape("&aHello &cWorld");
+    /// assert_eq!(str, "&_aHello &_cWorld");
+    /// ```
+    pub fn escape(&self, str: &str) -> String {
+        str.replace(self.marker, self.patterns.escaped())
     }
 
     /// Returns a string with all the possible formatting options.
     pub fn test(&self) -> String {
         self.convert(
-            "0123456789abcdefg"
+            &"0123456789abcdef"
                 .chars()
                 .map(|ch| format!("{m}{ch}{ch}", m = self.marker))
                 .chain(
-                    "lmno"
+                    "hijklmno"
                         .chars()
-                        .map(|ch| format!("{m}r{m}{ch}{ch}", m = self.marker)),
+                        .map(|ch| format!("{m}R{m}{ch}{ch}", m = self.marker)),
                 )
                 .collect::<String>(),
         )
-        .to_string()
+        .into_owned()
     }
```

Comment:
Same as `reset`, though the Python impl provided `python -m dahlia` in return 🤔

Reply:
Yh, it doesn't make much sense in a pure library and I don't want to add an executable just for this (cli is better handled by the sh version).